index.js

```diff
@@ -51,21 +59,29 @@ function arg(opts, {argv, permissive = false} = {}) {
 			}
 
 			if (!(argName in handlers)) {
-				if (permissive) {
+				const shortArg = argName.substring(0, 2);
+				const flags = argName.substring(1);
+				if (shortArg in handlers && flags.length > 1) {
+					for (const char of flags) {
+						if (`-${char}` in handlers) {
+							argv.push(`-${char}`);
+						} else {
+							throw new Error(`Unkown or unexpected option: -${char}`);
+						}
+					}
+
+					continue;
+				} else if (permissive) {
 					result._.push(arg);
 					continue;
 				} else {
-					const err = new Error(`Unknown or unexpected option: ${originalArgName}`);
-					err.code = 'ARG_UNKNOWN_OPTION';
```

Comment:
Ditching `ARG_UNKNOWN_OPTION` `code` would be a regression. This should make some test fail

Reply:
Totally glazed over that, thanks!